src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -50,16 +51,17 @@
     private String[] roomIds;
     private boolean replyBroadcast;
     private String responseString = null;
+    private Item item;
 
-    public StandardSlackService(String baseUrl, String teamDomain, String authTokenCredentialId, boolean botUser, String roomId) {
-        this(baseUrl, teamDomain, null, authTokenCredentialId, botUser, roomId, false);
+    public StandardSlackService(String baseUrl, String teamDomain, String authTokenCredentialId, boolean botUser, String roomId, Item item) {
```

Comment:
1. We can't modify existing constructors as this is part of the public API unless we are sure they aren't in use, normally you would deprecate first
2. I don't think we should be passing the Item down to this level, I haven't looked closely enough at the API but I would like to see a new constructor that passes a token that is guaranteed to be not null.
The legacy constructors will still need to be able to find a credential but that will be deprecated and removed at a later date

Reply:
Hopefully I addressed the 2 issues. I deprecated the other constructors, but maybe it's too early for that?